lnwallet/channel.go

```diff
@@ -4976,6 +4976,46 @@ func (lc *LightningChannel) AddHTLC(htlc *lnwire.UpdateAddHTLC,
 	return pd.HtlcIndex, nil
 }
 
+// GetDustSum takes in a boolean that determines which commitment to evaluate
+// the dust sum on. The return value is the sum of dust on the desired
+// commitment tx.
+//
+// NOTE: This over-estimates the dust exposure.
+func (lc *LightningChannel) GetDustSum(remote bool) lnwire.MilliSatoshi {
+	lc.RLock()
+	defer lc.RUnlock()
+
+	var dustSum lnwire.MilliSatoshi
+
+	dustLimit := lc.channelState.LocalChanCfg.DustLimit
+	if remote {
+		// Calculate dust sum on the remote's commitment.
+		dustLimit = lc.channelState.RemoteChanCfg.DustLimit
+	}
+
+	// Grab all of our HTLCs and evaluate against the dust limit.
+	for e := lc.localUpdateLog.Front(); e != nil; e = e.Next() {
+		pd := e.Value.(*PaymentDescriptor)
+		// If the satoshi amount is under the dust limit, add the msat
+		// amount to the dust sum.
+		if pd.EntryType == Add && pd.Amount.ToSatoshis() < dustLimit {
```

Comment:
Shouldn't this use the existing `isDust` method? Since it's possible that the HTLC is considered dust based on the fee rate and cost of the second level HTLC, but is actually _above_ this static dust value.

Reply:
The main reason I didn't do this was because it's prone to manipulation since you may have non-dust on the commitment (checks pass in the switch), the initiator sends a high update_fee and now the checks have been effectively bypassed.  It is fixed for zero-fee-htlc anchor channels.  I could add it, but it's still prone to error unless you use an estimate on some upper-bound feerate.  Could also add it to the mailbox, but the feerate itself isn't static unlike the dust params

Reply:
> while the mailbox uses the last sent or received (and possibly uncommitted) feerate which may lead to a slight discrepancy on what is dust.

I think _some_ slippage here is somewhat unavoidable, but in any case we'll always enforce that hard upper bound